cmake/utils.cmake

```
@@ -5,3 +5,18 @@ macro(ternary var boolean value1 value2)
         set(${var} ${value2})
     endif()
 endmacro()
+
+# Function to extract #define values from C/C++ headers
+function(extract_define_from_header header_file define_name output_var)
+    file(STRINGS "${header_file}" define_line REGEX "#define ${define_name}")
```

you might want to handle whitespace between `#define` and the name like this as well:
```suggestion
    file(STRINGS "${header_file}" define_line REGEX "#define[ \t]+${define_name}")
```

---

Done.